core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/function/RecordAggregateFunction.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch.aggregator.function;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+
+import javax.annotation.Nullable;
+
+/**
+ *
+ * @param <T> type of aggregate value
+ */
+public interface RecordAggregateFunction<T> extends AggregateFunction<T> {
```

Comment:
It also shouldn't extend AggregateFunction, because then you require that it does both selection and aggregation which are completely separate things.  If you just need the start, update, finish methods from AggregateFunction, you can create a common interface they both extend.